frontend/src/bundles/video_editor/components/timeline/subcomponents/timecursor.tsx

```diff
@@ -0,0 +1,118 @@
+import { useTimelineContext } from 'dnd-timeline';
+import { useCallback, useLayoutEffect, useRef, useState } from 'react';
+
+interface TimeCursorProperties {
+    interval?: number; 
+}
+
+const TimeCursor = (properties: TimeCursorProperties): JSX.Element | null => {
+    const timeCursorReference = useRef<HTMLDivElement>(null); 
+    const renderTimeReference = useRef(Date.now());
+    const { range, direction, sidebarWidth, valueToPixels, pixelsToValue } = useTimelineContext(); 
+
+    const side = direction === 'rtl' ? 'right' : 'left'; 
+
+    const isVisible = true;
+    const [isDragging, setIsDragging] = useState(false);
+    const [cursorPosition, setCursorPosition] = useState<number | null>(null); 
+
+    useLayoutEffect(() => {
+        const offsetCursor = (): void => {
+            if (!timeCursorReference.current || cursorPosition !== null) {
+                return;
+            }
+            const timeDelta = Date.now() - renderTimeReference.current; 
+            const timeDeltaInPixels = valueToPixels(timeDelta); 
+
+            const sideDelta = sidebarWidth + timeDeltaInPixels; 
+            timeCursorReference.current.style[side] = `${sideDelta}px`;
+        };
+        offsetCursor();
+        const interval = setInterval(offsetCursor, properties.interval || 1000);
+        return () => {
+            clearInterval(interval); 
+        };
+    }, [
+        side,
+        sidebarWidth,
+        properties.interval,
+        range.start,
+        valueToPixels,
+        isVisible,
```

Comment:
Why do you need isVisible variable?

Reply:
Initially I created it so the component could be collapsible but then I changed course and didt delete it. I'll delete it